### PR TITLE
Binary Tree Bottom View, Top View and Right View in C #1577

### LINKED ISSUE #1577: Top View, Left View, Right View, Bottom View of Binary Tree in C, Java, Python language
I would like to contribute on this

### PR DESCRIPTION
**Pull request by AkanshaKamboj**
Fixes #1577

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by asha15** [comment]
please include the issue number next to `Fixes #`

**Comment by AkanshaKamboj** [comment]
@asha15  @singh-shreya6 @MastersAbh Please review my PR.

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+};
+
+
+/* Inserting elements into Binary Tree*/
+struct Node *create()
+{
+    Node *ptr;
+    int x;
+    printf("Enter data(-1 for no node):");
+    scanf("%d",&x);
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Please add a space after ,

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    if(x == -1)
+    return NULL;
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Indentation

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    ptr = (struct Node*)malloc(sizeof(struct Node));
+    ptr->data = x;
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around ->

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    printf("Enter left child of %d:\n",x);
+    ptr->left = create();
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around ->

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    printf("Enter right child of %d:\n",x);
+    ptr->right = create();
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around ->

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        return;
+
+    if(level > a[50+dist])
+    {
+        a[50+dist] = level;
+        b[50+dist] = root->data;
+    }
+    
+    # Recursively calling the right and the left child to get the bottom view
+    findBottomView(root->right, a, b, level+1, dist+1);
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around ->

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+void findBottomView(struct Node *root, int *a, int *b, int level, int dist)
+{
+    if(root == NULL)
+        return;
+
+    if(level > a[50+dist])
+    {
+        a[50+dist] = level;
+        b[50+dist] = root->data;
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around ->

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+    }
+    
+    # Recursively calling the right and the left child to get the bottom view
+    findBottomView(root->right, a, b, level+1, dist+1);
+    findBottomView(root->left, a, b, level+1, dist-1);
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around ->

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    
+    int i,o=0;
+
+    int *a = (int *)calloc(100,sizeof(int));
+    int *b = (int *)calloc(100,sizeof(int));
+    int *c = (int *)calloc(100,sizeof(int));
+    
+    /* Initialising a(level of the node) and b(Vertical distance of the node from root) as INT_MIN as we need to print 
+       bottom most elements*/
+    for(i=0;i<100;i++)
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around the operator

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    findBottomView(root,a,b,0,0);
+    
+    for(i=0;i<100;i++)
+    {
+        if(b[i]!= INT_MIN)
+            c[o++]=b[i];
+    }
+
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around the operator

File: Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    return c;
+}
+
+int main()
+{
+    struct Node *root = create();
+    int len = 0;
+    int *res = bottomView(root, &len);
+    
+    for(int i=0; i<len; i++)
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/BinaryTree_Bottom_View.c]: Add space around the operator

**Comment by somya-kapoor** [comment]
@AkanshaKamboj Please the required in the whole file.

**Comment by somya-kapoor** [comment]
@AkanshaKamboj You have created 3 PRs for the same issue, please close the other 2 issues.

**Comment by AkanshaKamboj** [comment]
Made all the changes. Please review it now.

**Comment by asha15** [comment]
changed the file name BinaryTree_Bottom_View.c  into Binary_Tree_Bottom_View.c

**Comment by AkanshaKamboj** [comment]
Please review my PR. @asha15 @somya-kapoor

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+#include<limits.h>
+
+/* A binary tree node has data, pointer to left child and a pointer to right child */
+struct Node
+{
+    int data;
+    struct Node* left;
+    struct Node* right;
+};
+
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: remove this line

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+
+/* Inserting elements into Binary Tree*/
+struct Node *create()
+{
+    Node *ptr;
+    int x;
+    printf("Enter data(-1 for no node):");
+    scanf("%d",&x);
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: keep a space after comma

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+
+    if(x == -1)
+    return NULL;
+
+    ptr = (struct Node*)malloc(sizeof(struct Node));
+    ptr->data = x;
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: keep space before and after ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+    
+    printf("Enter left child of %d:\n", x);
+    ptr->left = create();
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: keep space before and after ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+    
+    printf("Enter right child of %d:\n", x);
+    ptr->right = create();
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: keep space before and after ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+    
+    return ptr;
+}
+
+/* Calculating the level of the node and the vertical distance of the node and 
+    finding the nodes that appear as the top view*/
+
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: remove this line

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+void findBottomView(struct Node *root, int *a, int *b, int level, int dist)
+{
+    if(root == NULL)
+        return;
+
+    if(level > a[50+dist])
+    {
+        a[50+dist] = level;
+        b[50+dist] = root->data;
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: keep space around operator

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    
+    # Recursively calling the right and the left child to get the bottom view
+    
+    findBottomView(root->right, a, b, level+1, dist+1);
+    
+    findBottomView(root->left, a, b, level+1, dist-1);
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: remove the extra lines

keep space around operators

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+int *bottomView(struct Node* root, int *len) {
+    
+    int i, o=0;
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: keep space around operators

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+
+    int *a = (int *)calloc(100,sizeof(int));
+    int *b = (int *)calloc(100,sizeof(int));
+    int *c = (int *)calloc(100,sizeof(int));
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: keep space after comm

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+    
+    /* Initialising a(level of the node) and b(Vertical distance of the node from root) as INT_MIN as we need to print 
+       bottom most elements*/
```
Comment (asha15) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.c]: one line should contain max 80 characters, change it according to it

**Comment by asha15** [comment]
Please remove the space after  # in next to fixes in the PR template. then we are good to merge 

Do this and let me know

**Comment by AkanshaKamboj** [comment]
Completed @asha15  and thank you

**Comment by asha15** [comment]
Good job :)

**Comment by AkanshaKamboj** [comment]
@asha15 Please look into my other PR as well. It's still pending #2395

**Pull request merged**